contracts/strategies/aura/base/AuraStrategyAuraBAL.sol

```diff
@@ -0,0 +1,624 @@
+    funds.sender = address(this);
+    funds.recipient = recipient;
+
+    int256[] memory limits = new int256[](batchSwapTokens);
+    limits[0] = int256(tokenBalance);
+
+    IBVault(bVault).batchSwap(
+      IBVault.SwapKind.GIVEN_IN,
+      swaps,
+      assets,
+      funds,
+      limits,
+      999999999999999999
+    );
+  }
+
+  // Deposit WETH to Balancer and Receive 80BAL20WETH BPT
+  // Deposit 80BAL20WETH to Aura: auraBALBpt Depositor and Receive auraBALBpt
+  function _convertCrvToAuraBAL() 
+    internal 
+  {
+    address universalRewardToken = rewardToken();
+    uint256 depositTokensAmount = depositNTokens();
+    uint256 tokenBalance = IERC20(universalRewardToken).balanceOf(address(this));
+
+    IERC20(universalRewardToken).safeApprove(bVault, 0);
+    IERC20(universalRewardToken).safeApprove(bVault, tokenBalance);
+
+    // we can accept 0 as minimum, this will be called only by trusted roles
+    
+    IAsset[] memory assets = new IAsset[](depositTokensAmount);
+    for (uint256 i = 0; i < depositTokensAmount; i++) {
+      assets[i] = IAsset(poolAssets[i]);
+    }
+
+    IBVault.JoinKind joinKind = IBVault.JoinKind.EXACT_TOKENS_IN_FOR_BPT_OUT;
+    uint256[] memory amountsIn = new uint256[](depositTokensAmount);
+    amountsIn[depositArrayIndex()] = tokenBalance;
+    uint256 minAmountOut = 1;
+
+    bytes memory userData = abi.encode(joinKind, amountsIn, minAmountOut);
+
+    IBVault.JoinPoolRequest memory request;
+    request.assets = assets;
+    request.maxAmountsIn = amountsIn;
+    request.userData = userData;
+    request.fromInternalBalance = false;
+
+    IBVault(bVault).joinPool(
+      balancerPoolId(),
+      address(this),
+      address(this),
+      request
+    );
+
+    address depositor = auraDeposit();
+    address universalDepositToken = depositToken();
+    uint256 bptBalance = IERC20(universalDepositToken).balanceOf(address(this));
+    IERC20(universalDepositToken).safeApprove(depositor, 0);
+    IERC20(universalDepositToken).safeApprove(depositor, bptBalance);
+    IAuraDepositor(depositor).deposit(bptBalance, true); //deposit and stake
+  }
+
+  /**   Stakes everything the strategy holds into the reward pool
+   */
+  function _investAllUnderlying() 
+    internal 
+    onlyNotPausedInvesting 
+  {
+    // this check is needed, because most of the SNX reward pools will revert if
+    // you try to stake(0).
+    if(IERC20(underlying()).balanceOf(address(this)) > 0) {
+      _enterRewardPool();
+    }
+  }
+
+  // Reward => Deposit(WETH) => JoinPool Bpt => Deposit(Bpt) => StakeAll
+  function _enterRewardPool() 
+    internal 
+  {
+    address lpToken = underlying();
+    address stakePool = rewardPool();
+    uint256 entireBalance = IERC20(lpToken).balanceOf(address(this));
+    IERC20(lpToken).safeApprove(stakePool, 0);
+    IERC20(lpToken).safeApprove(stakePool, entireBalance);
+    IAuraBaseRewardPool(stakePool).stakeAll(); // Stake
+  }
+
+  function addRewardToken(address _token) 
```

Comment:
I dont think this works. We can't add a reward token without also adding the liquidation info

Reply:
It should look more like this: https://github.com/harvest-finance/harvest-strategy/blob/master/contracts/strategies/convex/base/ConvexStrategy4Token.sol#L196

But then adding the path and dexes to `storedLiquidationPaths` and `storedLiquidationDexes`

Reply:
Fixed! Thank you for the suggestions. Indeed the original implementation won't liquidate the new reward token successfully.